### ISSUE TITLE
`data:` URL validation is too narrow.

### ISSUE THREAD

**Issue by puellanivis** [benmyrgorod/validate#2]
A non-exhaustive list of URIs that fail validation can be seen here: https://play.golang.org/p/pKsNZZ00AKo

Of particular note, most of the example `data:` URIs given in RFC-2397 do not validate, despite being specific explicitly valid data URIs.

**Comment by benmyrgorod** [comment]
Good catch! Looks like it is a similar issue to https://github.com/go-playground/validator/pull/548. Needs work.

**Comment by puellanivis** [comment]
Yeah, there was a poor regex that managed to spread pretty wide to nearly all of the validator packages. This is basically the canary test that I tend to do on every validate library that I find being used in our company. :laughing:
